includes/embeds/class-amp-tumblr-embed-handler.php

```diff
@@ -6,54 +6,89 @@
  */
 class AMP_Tumblr_Embed_Handler extends AMP_Base_Embed_Handler {
 
 	/**
-	 * Register embed.
+	 * Default width.
+	 *
+	 * Tumblr embeds for web have a fixed width of 540px.
+	 * See <https://tumblr.zendesk.com/hc/en-us/articles/226261028-Embed-pro-tips>.
+	 *
+	 * @var int
 	 */
-	public function register_embed() {
-		add_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10, 2 );
-	}
+	protected $DEFAULT_WIDTH = 540;
 
 	/**
-	 * Unregister embed.
+	 * Base URL used for identifying embeds.
+	 *
+	 * @var string
 	 */
-	public function unregister_embed() {
-		remove_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10 );
+	protected $base_embed_url = 'https://embed.tumblr.com/embed/post/';
+
+	/**
+	 * Get all raw embeds from the DOM.
+	 *
+	 * @param Document $dom Document.
+	 * @return DOMNodeList|null A list of DOMElement nodes, or null if not implemented.
+	 */
+	protected function get_raw_embed_nodes( Document $dom ) {
+		return $dom->xpath->query( sprintf( '//div[ @class = "tumblr-post" and starts-with( @data-href, "%s" ) ]', $this->base_embed_url ) );
 	}
 
 	/**
-	 * Filter oEmbed HTML for Tumblr to prepare it for AMP.
+	 * Make embed AMP compatible.
 	 *
-	 * @param string $cache Cache for oEmbed.
-	 * @param string $url   Embed URL.
-	 * @return string Embed.
+	 * @param DOMElement $node DOM element.
 	 */
-	public function filter_embed_oembed_html( $cache, $url ) {
-		$parsed_url = wp_parse_url( $url );
-		if ( false === strpos( $parsed_url['host'], 'tumblr.com' ) ) {
-			return $cache;
-		}
+	protected function sanitize_raw_embed( DOMElement $node ) {
+		$dom        = Document::fromNode( $node );
+		$iframe_src = $node->getAttribute( 'data-href' );
+
+		$attributes = [
+			'src'       => $iframe_src,
+			'layout'    => 'responsive',
+			'width'     => $this->args['width'],
+			'height'    => $this->args['height'],
+			'resizable' => '',
+			'sandbox'   => 'allow-scripts allow-popups allow-same-origin',
+		];
+
+		$amp_node = AMP_DOM_Utils::create_node(
+			$dom,
+			$this->amp_tag,
+			$attributes
+		);
 
-		// @todo The iframe will not get sized properly.
-		if ( preg_match( '#data-href="(?P<href>https://embed.tumblr.com/embed/post/\w+/\w+)"#', $cache, $matches ) ) {
-			$cache = AMP_HTML_Utils::build_tag(
-				'amp-iframe',
-				[
-					'width'   => $this->args['width'],
-					'height'  => $this->args['height'],
-					'layout'  => 'responsive',
-					'sandbox' => 'allow-scripts allow-popups', // The allow-scripts is needed to allow the iframe to render; allow-popups needed to allow clicking.
-					'src'     => $matches['href'],
-				],
-				sprintf( '<a placeholder href="%s">Tumblr</a>', $url )
-			);
+		// Add an overflow node to allow the amp-iframe to resize.
+		$overflow_node = AMP_DOM_Utils::create_node(
+			$dom,
+			'div',
+			[
+				'overflow'   => '',
+				'tabindex'   => 0,
+				'role'       => 'button',
+				'aria-label' => esc_attr__( 'See more', 'amp' ),
```

Comment:
No need to escape when manipulating the DOM:

```suggestion
				'aria-label' => __( 'See more', 'amp' ),
```